File: solutions/Leetcode_1/leetcode_1.py
from typing import List


class Solution1:
    def two_sum(self, nums: List[int], target: int) -> List[int]:
        num_idx = {}
        for i, x in enumerate(nums):
            diff = target - x
            if num_idx.get(diff, None) is not None:
                return [i, num_idx[diff]]
            else:
                num_idx[x] = i

File: solutions/Leetcode_1/leetcode_1.cpp
#include <unordered_map>
#include <vector>

class Solution1
{
public:
    std::vector<int> twoSum(std::vector<int> &nums, int target)
    {
        std::unordered_map<int, int> num_idx;
        for (int i = 0; i < nums.size(); i++)
        {
            int diff = target - nums[i];
            if (num_idx.find(diff) != num_idx.end())
                return {i, num_idx[diff]};
            else
                num_idx[nums[i]] = i;
        }
        return {};
    }
};